File: application/app/src/test/java/com/yahyeet/boardbook/GameTests.java
package com.yahyeet.boardbook;

import com.yahyeet.boardbook.model.entity.Game;
import com.yahyeet.boardbook.model.entity.GameRole;
import com.yahyeet.boardbook.model.entity.GameTeam;

import org.junit.Test;

import java.util.ArrayList;
import java.util.List;

import static org.junit.Assert.*;

public class GameTests {

    public GameTests(){}

    //TODO Write test for seeing if name,desc.... are correct, doesn't seem very important


    /// See if you can add a team
    @Test
    public void addTeam(){
        Game game = getTestGame();
        game.getTeams().add(new GameTeam("SOM"));
        assertEquals( "SOM", game.getTeams().get(0).getName());
        game.getTeams().add(new GameTeam("LSOA"));
        assertEquals( "LSOA", game.getTeams().get(1).getName());
    }

    /// See if you can add roles to a team
    @Test
    public void addRole(){
        Game game = getTestGame();

        GameTeam lsoaTeam = new GameTeam("LSOA");
        lsoaTeam.getRoles().add(new GameRole("Merlin"));

        assertEquals( "Merlin", game.getTeams().get(0).getRoles().get(0).getName());

        GameTeam somTeam = new GameTeam("SOM");
        somTeam.getRoles().add(new GameRole("Mordred"));
        assertEquals( "Mordred", game.getTeams().get(0).getRoles().get(0).getName());
    }

    private Game getTestGame(){
        return new Game("Avalon", "Descriptive Description", 3, 2, 7);
    }

}

File: application/app/src/test/java/com/yahyeet/boardbook/GameTests.kt
package com.yahyeet.boardbook

import com.yahyeet.boardbook.model.entity.Game
import com.yahyeet.boardbook.model.entity.GameRole
import com.yahyeet.boardbook.model.entity.GameTeam

import org.junit.Test

import org.junit.Assert.*

class GameTests {

    private val testGame: Game
        get() = Game("Avalon", "Descriptive Description", 3, 2, 7)

    //TODO Write test for seeing if name,desc.... are correct, doesn't seem very important


    /// See if you can add a team
    @Test
    fun addTeam() {
        val game = testGame
        game.teams.add(GameTeam("SOM"))
        assertEquals("SOM", game.teams[0].name)
        game.teams.add(GameTeam("LSOA"))
        assertEquals("LSOA", game.teams[1].name)
    }

    /// See if you can add roles to a team
    @Test
    fun addRole() {
        val game = testGame

        val lsoaTeam = GameTeam("LSOA")
        lsoaTeam.roles.add(GameRole("Merlin"))
        game.teams.add(lsoaTeam)

        assertEquals("Merlin", game.teams[0].roles[0].name)

        val somTeam = GameTeam("SOM")
        somTeam.roles.add(GameRole("Mordred"))
        game.teams.add(somTeam)
        assertEquals("Mordred", game.teams[1].roles[0].name)
    }

}

